src/main/java/nl/tudelft/contextproject/model/level/MSTBasedLevelFactory.java

```diff
@@ -0,0 +1,332 @@
+		RoomNode treasureNode = new RoomNode(startAndEndRooms.getTreasureRoom(), TREASURE_ROOM_ID);
+		addRoomNode(treasureNode, treasureLocation);
+	}
+
+	/**
+	 * Place the other rooms.
+	 * Will try to place random rooms until max attempts has been reached or there are no rooms left to place.
+	 */
+	public void placeOtherRooms() {
+		int attempts = 0;
+		int randomIndex;
+		RoomNode currentNode;
+		while (attempts <= MAX_ATTEMPTS && !baseNodes.isEmpty()) {
+
+			//Get random room
+			randomIndex = rand.nextInt(baseNodes.size());
+			currentNode = baseNodes.get(randomIndex);
+
+			//Get random coordinates
+			Vec2I coordinates = new Vec2I(RandomUtil.getRandomIntegerFromInterval(rand, 
+					RoomNode.MIN_DIST, MAX_WIDTH - (currentNode.room.size.getWidth() + RoomNode.MIN_DIST + 1)), 
+					RandomUtil.getRandomIntegerFromInterval(rand, 
+							RoomNode.MIN_DIST, MAX_HEIGHT - (currentNode.room.size.getHeight() + RoomNode.MIN_DIST + 1)));
+
+			//Check placement and place if possible
+			if (currentNode.scanPossiblePlacement(mazeTiles, coordinates)) {
+				addRoomNode(currentNode, coordinates);
+			}
+			attempts++;
+		}
+	}
+
+	/**
+	 * Create edges between all RoomNodes.
+	 * Creates edges from exits from one room to entrances to another room.
+	 * It creates an edge from every exit to every entrance.
+	 */
+	public void createEdges() {
+		int corridorIdcounter = 0;
+		for (RoomEntrancePoint entrance : usedEntrancePoints) {
+			for (RoomExitPoint exit : usedExitPoints) {
+				if (!entrance.node.equals(exit.node)) {
+					CorridorEdge edge = new CorridorEdge(exit, entrance, corridorIdcounter++);
+					edges.put(edge.id, edge);
+					exit.node.addOutgoingEdge(edge);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Create a MST, connecting all the rooms with the least amount of path.
+	 * Create the MST object.
+	 * Run the MST algorithm, selecting which edges are chosen in map generation.
+	 * 
+	 */
+	protected void createReverseMST() {
+		MinimumSpanningTree mst = new MinimumSpanningTree(usedNodes, edges);
+		mst.runReverseDeleteAlgorithm();
+		chosenEdges = mst.getCorridorIDsReverseAlgorithm();
+	}
+
+	/**
+	 * Place the corridors from the chosenEdges list.
+	 * For each corridor a breadth first search is done from
+	 * the corridor starting point to the corridor end point.
+	 * This progress creates a list of stacks on that contain locations that should
+	 * be turned into a corridor tile.
+	 */
+	protected void placeCorridors() {
+		ArrayList<Stack<Vec2I>> corridorList = getCorridorLocations();
+		Vec2I carveLocation;
+		int x, y;
+		for (Stack<Vec2I> stack : corridorList) {
+			while (!stack.empty()) {
+				carveLocation = stack.pop();
+				x = carveLocation.x;
+				y = carveLocation.y;
+				if (mazeTiles[x][y] == null) {
+					mazeTiles[x][y] = new MazeTile(x, y, TileType.CORRIDOR);
+				}
+			}
+		}
+	}	
+
+	/**
+	 * Get all corridor locations on the map.
```

Comment:
please add a newline after this line, so the @ lines are seperated from the method description